corehq/apps/receiverwrapper/views.py

```diff
@@ -73,43 +73,89 @@ def _process_form(request, domain, app_id, user_id, authenticated,
-        })
-        return _submission_error(
-            request, "Received a submission with POST.keys()", metric_tags,
-            domain, app_id, user_id, authenticated, meta,
-        )
+        response = _multimedia_bug_response(request, domain, authenticated,
+                                            metric_tags, app_id, user_id)
+    else:
+        if isinstance(instance, BadRequest):
+            response = _bad_request_response(instance.message, metric_tags)
+        # silently ignore submission if it meets ignore-criteria
+        elif should_ignore_submission(instance, request):
+            response = _ignored_submission_response(metric_tags)
+        elif toggles.FORM_SUBMISSION_BLACKLIST.enabled(domain):
+            response = _blacklisted_domain_response(metric_tags)
+    return instance, attachments, response
 
-    if isinstance(instance, BadRequest):
-        response = HttpResponseBadRequest(instance.message)
-        _record_metrics(metric_tags, 'known_failures', response)
-        return response
 
-    if should_ignore_submission(request):
-        # silently ignore submission if it meets ignore-criteria
-        response = openrosa_response.SUBMISSION_IGNORED_RESPONSE
-        _record_metrics(metric_tags, 'ignored', response)
-        return response
+def _multimedia_bug_response(request, domain, authenticated, metric_tags,
+                             app_id, user_id):
+    try:
+        instance = request.FILES[MAGIC_PROPERTY].read()
+        xform = convert_xform_to_json(instance)
+        meta = xform.get("meta", {})
+    except:
+        meta = {}
+
+    metrics_counter('commcare.corrupt_multimedia_submissions', tags={
+        'domain': domain, 'authenticated': authenticated
+    })
+    return _submission_error(
+        request, "Received a submission with POST.keys()", metric_tags,
+        domain, app_id, user_id, authenticated, meta,
+    )
 
-    if toggles.FORM_SUBMISSION_BLACKLIST.enabled(domain):
-        response = openrosa_response.BLACKLISTED_RESPONSE
-        _record_metrics(metric_tags, 'blacklisted', response)
-        return response
 
-    with TimingContext() as timer:
-        app_id, build_id = get_app_and_build_ids(domain, app_id)
-        submission_post = SubmissionPost(
+def _bad_request_response(message, metric_tags):
+    response = HttpResponseBadRequest(message)
+    _record_metrics(metric_tags, 'known_failures', response)
+    return response
+
+
+def _ignored_submission_response(metric_tags):
+    response = openrosa_response.SUBMISSION_IGNORED_RESPONSE
+    _record_metrics(metric_tags, 'ignored', response)
+    return response
+
+
+def _blacklisted_domain_response(metric_tags):
+    response = openrosa_response.BLACKLISTED_RESPONSE
+    _record_metrics(metric_tags, 'blacklisted', response)
+    return response
+
+
+def _submission_post_object(*, instance, attachments, domain, app_id, build_id, auth_cls, user_id, authenticated,
```

Comment:
what's the reason to force these to be keyword args?

Reply:
this would just help ensure, where its used, we always pass keyword args considering the number of arguments and not mess it up, which I thought was a good thing to do, unless we can reduce the number of arguments getting passed in future.

Did it as a part of https://github.com/dimagi/commcare-hq/pull/28424#discussion_r476691586